### ISSUE TITLE
Snakecharm keeps crashing

### ISSUE THREAD

**Issue by winni2k** [JetBrains-Research/snakecharm#168]
I keep getting this error message when using Snakecharm version `0.4.42.2018.3`

![image](https://user-images.githubusercontent.com/304958/62471772-a5bebc80-b79d-11e9-8a25-f233f34876e5.png)

This is the first error message from the box:

```
java.lang.NoSuchMethodException: com.jetbrains.python.parsing.ExpressionParsing.parseFormattedStringNode()
	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2475)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.parseMultilineString(SnakemakeExpressionParsing.kt:314)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.parseStringAdditiveExpression(SnakemakeExpressionParsing.kt:412)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.parseSingleExpressionOrStringExpression(SnakemakeExpressionParsing.kt:247)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.parseRuleParamArgument(SnakemakeExpressionParsing.kt:238)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.access$parseRuleParamArgument(SnakemakeExpressionParsing.kt:21)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing$parseRuleLikeSectionArgumentList$1.invoke(SnakemakeExpressionParsing.kt:36)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing$parseRuleLikeSectionArgumentList$1.invoke(SnakemakeExpressionParsing.kt:21)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.parseArgumentAndReportErrors(SnakemakeExpressionParsing.kt:209)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.access$parseArgumentAndReportErrors(SnakemakeExpressionParsing.kt:21)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing$parseArgumentList$result$1.invoke(SnakemakeExpressionParsing.kt:48)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing$parseArgumentList$result$1.invoke(SnakemakeExpressionParsing.kt:21)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.doParseRuleParamArgumentList(SnakemakeExpressionParsing.kt:183)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.parseArgumentList(SnakemakeExpressionParsing.kt:47)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeExpressionParsing.parseRuleLikeSectionArgumentList(SnakemakeExpressionParsing.kt:33)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeStatementParsing.parseRuleParameter(SnakemakeStatementParsing.kt:266)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeStatementParsing.parseRuleLikeDeclaration(SnakemakeStatementParsing.kt:206)
	at com.jetbrains.snakecharm.lang.parser.SnakemakeStatementParsing.parseStatement(SnakemakeStatementParsing.kt:107)
	at com.jetbrains.python.parsing.PyParser.parse(PyParser.java:44)
	at com.jetbrains.python.psi.PyFileElementType.parseContents(PyFileElementType.java:95)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.lambda$ensureParsed$0(LazyParseableElement.java:198)
	at com.intellij.psi.impl.DebugUtil.performPsiModification(DebugUtil.java:564)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.ensureParsed(LazyParseableElement.java:197)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:246)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:40)
	at com.intellij.lang.folding.CustomFoldingBuilder.addCustomFoldingRegionsRecursively(CustomFoldingBuilder.java:78)
	at com.intellij.lang.folding.CustomFoldingBuilder.buildFoldRegions(CustomFoldingBuilder.java:42)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptorsNoPlaceholderCaching(LanguageFolding.java:90)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:72)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:233)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:213)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.buildInitialFoldings(CodeFoldingManagerImpl.java:191)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorImpl.loadEditorInBackground(PsiAwareTextEditorImpl.java:50)
	at com.intellij.openapi.fileEditor.impl.text.AsyncEditorLoader.lambda$null$0(AsyncEditorLoader.java:88)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:704)
	at com.intellij.openapi.fileEditor.impl.text.AsyncEditorLoader.lambda$scheduleLoading$1(AsyncEditorLoader.java:88)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.insideReadAction(NonBlockingReadActionImpl.java:162)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$null$2(NonBlockingReadActionImpl.java:140)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1106)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:72)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:126)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:115)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:72)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$6(NonBlockingReadActionImpl.java:140)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:207)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.lambda$run$0(BoundedTaskExecutor.java:185)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:208)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:181)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by iromeo** [comment]
Which IDE version do you use?

**Comment by iromeo** [comment]
[snakecharm-0.4.44.2018.3.zip](https://github.com/JetBrains-Research/snakecharm/files/3469474/snakecharm-0.4.44.2018.3.zip)
@winni2k Please try this version, to install plugin from zip file read https://github.com/JetBrains-Research/snakecharm#install-plugin-from-zip-bundle

**Comment by winni2k** [comment]
I will try the provided plugin zip.

Also, my PyCharm version info is as follows:

```
PyCharm 2019.2 (Professional Edition)
Build #PY-192.5728.105, built on July 23, 2019
Licensed to warren kretzschmar
Subscription is active until April 17, 2020
For educational use only.
Runtime version: 11.0.3+12-b304.10 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.5
GC: ParNew, ConcurrentMarkSweep
Memory: 1979M
Cores: 12
Registry: ide.tooltip.initialDelay=600
Non-Bundled Plugins: R4Intellij, com.tylerthrailkill.intellij.solarized, name.kropp.intellij.makefile, net.seesharpsoft.intellij.plugins.csv, BashSupport
```

**Comment by winni2k** [comment]
Looks good so far. I'll reopen this issue if I get any further crashes.

**Comment by iromeo** [comment]
Thanks, we will update the plugin in the Plugins Repository soon.

**Comment by iromeo** [comment]
@winni2k Thanks for the bug report, the issue affects only non-community editions of PyCharm, IntelliJ IDEA, etc.

Bug fix update released: Version 0.4 build #44
